controller/src/beerocks/master/wfa_ca.cpp

```diff
@@ -676,48 +678,26 @@ void wfa_ca::handle_wfa_ca_message(
 }
 
 /**
- * @brief Create CMDU to send when receiving WFA-CA command "DEV_SEND_1905".
- * This function may create the CMDU from scratch or use the certification buffer which loaded with
- * preperared from advance.
+ * @brief Create CMDU manually by loading CMDU from preperared buffer, or getting the data to fill 
+ * TLVs from the database.
+ * If the function fails but err_string string is empty, then it means that the CMDU was not handled
+ * by this function, and the caller will need to create the CMDU by itself.
  * 
  * @param[in,out] cmdu_tx CmduTx message object created from certification_tx_buffer.
- * @param[in] message_type CMDU message type to create. 
- * @return true if successful, false if not.
+ * @param[in] message_type CMDU message type to create.
+ * @param[in] database Controllers database.
+ * @param[out] err_string Contains an error description if the function fails.
+ * @return true if CMDU was successfully loaded, false with empty err_string if CMDU was not loaded,
+ * but didn't fail, false with non-empty err_string on failure.
  */
-bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessageType message_type)
+bool wfa_ca::create_cmdu(ieee1905_1::CmduMessageTx &cmdu_tx, ieee1905_1::eMessageType message_type,
+                         db &database, std::string &err_string)
```

Comment:
You don't actually use the db argument...

But I guess this is a cherry-pick issue due to some other change you're making, so I'll mark that as (nitpick).

Reply:
It is preparation for taking data from the database as we talked today.
I mention it on the @breif.